armi/utils/tests/test_gridGui.py

```diff
@@ -0,0 +1,167 @@
+    armi.configure()
+from armi.utils import gridEditor
+
+_SECONDS_PER_TICK = 0.01
+
+
+def _wait(num_ticks: int) -> None:
+    time.sleep(num_ticks * _SECONDS_PER_TICK)
+
+
+def _findPointInWindow(
+    window: wx.Window, offsetFromLeft: float = 0.5, offsetFromTop: float = 0.5
+) -> wx.Point:
+    """Given a window, return a point in it. Defaults to the center of the window object.
+
+    If offsets are smaller than 0 or greater than 1, this would return a point outside the window object.
+    """
+    rect: wx.Rect = window.GetScreenRect()
+    x = rect.x + int(offsetFromLeft * rect.width)
+    if x == rect.x + rect.width:
+        x = rect.x + rect.width - 1
+    y = rect.y + int(offsetFromTop * rect.height)
+    if y == rect.y + rect.height:
+        y = rect.y + rect.height - 1
+    return wx.Point(x, y)
+
+
+class GuiTestCase(unittest.TestCase):
+    """Provides scaffolding for a GUI test.
+
+    Without this scaffolding, the GUI's main loop would block the UIActionSimulator. Thus, the simulated actions and
+    asserts must be run asynchronously within the GUI's event loop. Since the asserts are also run asynchronously,
+    we need to make sure that the test does not end until all assert statements have been called, and that the test
+    outputs are properly passed to the test framework. The app is also properly torn down after each test.
+
+    This way, the user only needs to define the simulated actions and the expected behavior in order to write a UI test.
+    """
+
+    def initializeGui(self):
+        """The user can override this to initialize the GUI differently.
+
+        Note: This method is called in self.run(), before super().run. We deliberately avoid naming this 'setUp',
+        because super().run internally calls self.setUp, which would be too late.
+        """
+
+        self.app = wx.App()
+        self.frame = wx.Frame(
+            None, wx.ID_ANY, title="Grid Blueprints UI", pos=(0, 0), size=(1000, 1000)
+        )
+        self.gui = gridEditor.GridBlueprintControl(self.frame)
+        self.frame.Show()
+        self.inputSimulator = wx.UIActionSimulator()
+
+    def _cleanUpApp(self):
+        for window in wx.GetTopLevelWindows():
+            try:
+                assert window.IsModal()
+            except (AttributeError, AssertionError):
+                window and window.Close()
+            else:
+                window.EndModal(0)
+                self.app.ScheduleForDestruction(window)
+
+    def _runAsync(self, result):
+        super().run(result)
+        self._cleanUpApp()
+        self._testCompleted.set_result(None)
+
+    def run(self, result=None):
+        """Overrides unittest.TestCase.run."""
+        self.initializeGui()
+        loop = asyncio.get_event_loop()
+        self._testCompleted = loop.create_future()
+        wx.CallLater(0, self._runAsync, result)
+        self.app.MainLoop()
+        loop.run_until_complete(self._testCompleted)
+        return result
+
+
```

Comment:
Since these tests are a bit touchy, we may just want to make them opt-in?
Something like this seems to work well:
```
@pytest.mark.skipif(not bool(os.environ.get("ARMI_GUI_TESTS", False)), reason="GUI tests require a rather specific environment (see above), so these tests are opt-in")
```

Reply:
Done.